integrations/gitlab/gitlab_integration/events/hooks/push.py

```diff
@@ -52,8 +101,78 @@ async def _on_hook(self, body: dict[str, Any], gitlab_project: Project) -> None:
                 gitlab_project
             )
             await ocean.register_raw(ObjectKind.PROJECT, [enriched_project])
+
         else:
             logger.debug(
                 f"Skipping push hook for project {gitlab_project.path_with_namespace} because the ref {ref} "
                 f"does not match the branch {branch}"
             )
+
+    async def _process_files(
+        self,
+        gitlab_project: Project,
+        files: list[str],
+        spec_path: list[str],
+        commit_before: str,
+        commit_after: str,
+        branch: str,
+        file_action: FileAction,
+    ) -> None:
+        if not files:
+            return
+        logger.info(
+            f"Processing {file_action} files {files} for project {gitlab_project.path_with_namespace}"
+        )
+
+        for file in files:
+            try:
+                if does_pattern_apply(spec_path, file):
+                    logger.info(
+                        f"Found file {file} in spec_path {spec_path} pattern, processing its entity diff"
+                    )
```

Comment:
I wonder why would we want to pass all the files that have been added / removed / modified to this function and to this log.

it can be very spammy and irrelevant. 
I would rather move this outside of this method, to where we filter the each type of changed files.

Reply:
<img width="1173" alt="Screenshot 2024-09-18 at 12 16 37 PM" src="https://github.com/user-attachments/assets/a0a1b767-4651-4def-bde4-e80a3179a867">
So far I don't think it's spammy, we only log the relevant info that needs to be displayed. And this will be crucial for debugging